test/toolTests.ts

```diff
@@ -81,24 +81,83 @@ describe('Tool Tests', function () {
         });
     });
 
-    it('has status code in exception dictionary for non 200 response', function () {
-        this.timeout(2000);
 
-        return new Promise<void>(async(resolve, reject)=> {
-            try {
-                let nonExistentUrl: string = "http://httpbin.org/nonexistent";
-                let downPath: string = await toolLib.downloadTool(nonExistentUrl);
+    
+    describe('has status code in exception dictionary for HTTP error code responses', function () {
+        this.timeout(2000);
+        
+        let httpErrorCodes:number[] = [400,401,402,403,404,405,406,407,408,409,410,411,412,413,414,415,416,417,418,422,428,429,431,451,500,501,502,503,504,505,511,520,522,524]
+
+        for (let errorCodes of httpErrorCodes) {
+            it('error thrown on HTTP status code ' + errorCodes, async function() {
+                return new Promise<void>(async(resolve, reject)=> {
+                    try {
+                        let errorCodeUrl: string = `https://httpbin.org/status/${errorCodes}`;
+                        let downPath: string = await toolLib.downloadTool(errorCodeUrl);
+        
+                        reject('a file was downloaded but it shouldnt have been');
+                    } 
+                    catch (err){
+                        assert.equal(err['httpStatusCode'], errorCodes, 'status code exists');
+        
+                        resolve();
+                    }
+                });
+            });
+        }
 
-                reject('a file was downloaded but it shouldnt have been');
-            } 
-            catch (err){
-                assert.equal(err['httpStatusCode'], 404, 'status code exists');
 
-                resolve();
-            }
-        });
+        
     });
 
+    describe('accepts any status code < 400', function() {
```

Comment:
Can remove this test.

Reply:
Or modify to what Bryan said, check that for a redirect URL we get the final URL and a 200.